### PR TITLE
Testing react-aria implementation

### PR DESCRIPTION
**Pull request by Oluane**
## Description

This PR is a POC of react-aria implementation to help us improve the app's accessibility. 

First, the lib provides a special SSR hook, to ensure consistency between what server and client side. 
> Wrapping your application in an SSRProvider helps ensure that the HTML generated on the server matches the DOM structure hydrated on the client. 

So I wrapped the entire app in this hook in `_app.tsx`.

For first tests, I choose to add the hooks provided by the lib on two kind of elements : 
- Links
- Buttons

It's pretty hard for me to see the benefits that this brings for now, at least on those elements. (I'm not sure at the moment on what I need to look for etc..) 
However, I noticed an improvement on our button elements, some of them where not correctly tagged as "interactive" elements before, and they are now. I also add the "type" attribute on all of them.

I also made some tests with `input` elements, and the differences before/after was more obvious(but implementation on these elements needs more big changes, and should be made on another PR).

## Type of change

- [ ] **Chore** (non-breaking change which refactors / improves the existing code base).
- [ ] **Bug fix** (non-breaking change which fixes an issue).
- [ ] **New feature** (non-breaking change which adds functionality).
- [x] **Breaking change** (fix or feature that would cause existing functionality to not work as expected).

## List of added dependencies

We have choice on the lib management: 

All of hooks are available in separate packages, or in a bundle (100ko) that regroups all of them. 

I made the choice to add only the ones I used in this PR, but this choice needs to be discussed. 

So for now, I add: 

- @react-aria/ssr
- @react-aria/link
- @react-aria/button

## Checklist:

- [x] My code follows the style [**contributing guidelines**][contributing_file] of this project.
- [x] I have performed a self-review of my own code.
- [ ] I have commented my code in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] I have added tests that prove my fix is effective or that my feature works.

[contributing_file]: https://github.com/fewlinesco/connect-account/blob/master/README.adoc

### CODE REVIEW

**Comment by Fenntasy** [comment]
Task linked: [CU-fd7gtx [test] - Test react-aria](https://app.clickup.com/t/fd7gtx)

**Comment by Fuzznimp** [comment]
Good for me. I would like to have Vincent's input if we really need the lib, as you are unsure what it really adds.

**Comment by Fenntasy** [comment]
> (I'm not sure at the moment on what I need to look for etc..)
> However, I noticed an improvement on our button elements, some of them where not correctly tagged as "interactive" elements before, and they are now. I also add the "type" attribute on all of them.

What are the attributes for buttons and links on this PR?

File: src/components/buttons/buttons.tsx [hunk 1]
```diff
@@ -11,10 +12,24 @@ enum ButtonVariant {
 }
 
 interface ButtonProps {
+  type: "button" | "submit" | "reset";
```
[the remark this answers is not in the document]
Comment (Oluane) [reply src/components/buttons/buttons.tsx:15]: image

**Review by Fuzznimp — APPROVED** [review]

**Pull request merged**